### PR TITLE
handlebars: Improve MustacheStatement printing

### PR DESCRIPTION
**Pull request by dcyriller**
Here is the issue addressed by this PR:
```
{{!-- input --}}
<p>Hi here is your name, as it will be displayed  {{firstName}}  {{lastName}}   , welcome!</p>

{{!-- stable --}}
<p>
  Hi here is your name, as it will be displayed {{firstName}} {{lastName
  }} , welcome!
</p>

{{!-- master --}}
<p>
  Hi here is your name, as it will be displayed {{firstName}} {{
    lastName
  }} , welcome!
</p>
```

As a result, it will implement https://github.com/prettier/prettier/pull/7051#discussion_r351894842 and https://github.com/prettier/prettier/pull/7051#discussion_r351877770

It depends on [this first](https://github.com/prettier/prettier/pull/7072) and [second](https://github.com/prettier/prettier/pull/7072) PRs (they should be settled first).

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: changelog_unreleased/handlebars/pr-7157.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  Hi {{firstName}} {{lastName
+  }} , welcome!
+</p>
+
+{{!-- master --}}
+<p>
+  Hi {{firstName}} {{
+    lastName
+  }} , welcome!
+</p>
```
Comment (evilebottnawi) [inline changelog_unreleased/handlebars/pr-7157.md]: Can we add example without `width: 40`?

Comment (dcyriller) [reply changelog_unreleased/handlebars/pr-7157.md]: Sure 👍

Comment (dcyriller) [reply changelog_unreleased/handlebars/pr-7157.md]: I've updated the PR's description + the CHANGELOG to take a more regular example.

**Comment by dcyriller** [comment]
@evilebottnawi this is the next one. I just rebased it on top of master!

**Comment by dcyriller** [comment]
Thank you @lipis and @evilebottnawi 🙏

**Comment by lipis** [comment]
Thank you! :)

**Pull request merged**